### PR TITLE
Special case for order 1 zonotopes

### PR DESCRIPTION
**Pull request by mforets**
This PR makes two changes:

1) speedup computation of vertices of a 2d zonotope of order 1, it goes from ~2us (master) to ~500ns (this branch). (*)

2) fix the `_vertices_list_iterative` such that the type of the vertices corresponds to the type of the center of the zonotope. before, it was hardcoding `Vector{N}`, which is a bad idea that prevents from obtaining an SArray output if the input is SArray.

(*) Using static arrays, runs in ~60ns.

cc @yupbank

### CODE REVIEW

File: src/Interfaces/AbstractZonotope.jl [hunk 1]
```diff
@@ -426,7 +426,7 @@ this method; otherwise, redundant vertices may be present.
 function vertices_list(Z::AbstractZonotope{N};
                        apply_convex_hull::Bool=true) where {N<:Real}
     c = center(Z)
-    G = remove_zero_columns(genmat(Z))
+    G = genmat(Z)
```
Comment (schillic) [inline src/Interfaces/AbstractZonotope.jl:429]: This changes the result for `apply_convex_hull=false` in 2D. What is the motivation for this change? I do not see why 2D should behave differently.

```julia
julia> Z1 = Zonotope([0.0, 0.0], zeros(2, 1));
julia> Z2 = Zonotope([0.0, 0.0], zeros(2, 2));
julia> Z3 = Zonotope([0.0, 0.0], zeros(2, 4));

# this branch
julia> vertices_list(Z1; apply_convex_hull=false)
2-element Array{Array{Float64,1},1}:
 [0.0, 0.0]
 [0.0, 0.0]

julia> vertices_list(Z2; apply_convex_hull=false)
4-element Array{Array{Float64,1},1}:
 [0.0, 0.0]
 [0.0, 0.0]
 [0.0, 0.0]
 [0.0, 0.0]

julia> vertices_list(Z3; apply_convex_hull=false)
8-element Array{Array{Float64,1},1}:
 [0.0, 0.0]
 [0.0, 0.0]
 [0.0, 0.0]
 [0.0, 0.0]
 [0.0, 0.0]
 [0.0, 0.0]
 [0.0, 0.0]
 [0.0, 0.0]

# master
julia> vertices_list(Z1; apply_convex_hull=false)
1-element Array{Array{Float64,1},1}:
 [0.0, 0.0]
julia> vertices_list(Z2; apply_convex_hull=false)
1-element Array{Array{Float64,1},1}:
 [0.0, 0.0]
julia> vertices_list(Z3; apply_convex_hull=false)
1-element Array{Array{Float64,1},1}:
 [0.0, 0.0]
```

Comment (mforets) [reply src/Interfaces/AbstractZonotope.jl:429]: the motivation is for performance reasons, because remove_zero_columns doesn't need to be applied in general; in 2d, the convex hull will reduce such vertices efficiently.

**Review by schillic — APPROVED** [review]
I did not look at the build error...

**Comment by mforets** [comment]
the build error was a doctest that was failing, and i;ve sent the correction now (tests pass locally).  

in fact, the returned array is "better" in the sense that the points are sorted in CCW order, while in master it is not the case. (didn't actually check why, since i would have also expected that they are sorted).
